Sources/ScipioKit/Resolver/ResolvedModels.swift

```diff
@@ -0,0 +1,233 @@
+import Foundation
+import PackageManifestKit
+
+typealias Pins = [PackageID: Pin.State]
+
+struct ModulesGraph {
+    var rootPackage: ResolvedPackage
+    var allPackages: [PackageID: ResolvedPackage]
+    var allModules: Set<ResolvedModule>
+
+    func package(for module: ResolvedModule) -> ResolvedPackage? {
+        allPackages[module.packageID]
+    }
+
+    func module(for name: String) -> ResolvedModule? {
+        allModules.first { $0.name == name }
+    }
+}
+
+struct PackageID: Hashable {
+    var description: String
+    var packageIdentity: String
+
+    init(
+        packageKind: PackageKind,
+        packageIdentity: String
+    ) {
+        self.packageIdentity = packageIdentity
+        switch packageKind {
+        case .root(let url), .fileSystem(let url), .localSourceControl(let url):
+            description = url.absoluteString
+        case .remoteSourceControl(let string):
+            description = string
+        case .registry(let string):
+            description = string
+        }
+    }
+}
+
+struct ResolvedPackage: Identifiable {
```

Comment:
Should be Sendable?

Reply:
Since ResolvedPackage is internal, isn't it implicitly Sendable?